google/cloud/grpc_utils/internal/completion_queue_impl.cc

```diff
@@ -46,6 +45,8 @@ void CompletionQueueImpl::Run(CompletionQueue& cq) {
     if (op->Notify(cq, ok)) {
       ForgetOperation(tag);
     }
+    std::unique_lock<std::mutex> lk(mu_);
```

Comment:
Given that we now need the mutex to protect `pending_ops_.empty()`, might I suggest that `shutdown_` become a plain `bool` protected by the same mutex (i.e., get rid of the use of `<atomic>`).

Reply:
Great idea, thanks.